src/pexprs-madlibs.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+pexprs.PExpr.prototype.generateExample = common.abstract;
+
+pexprs.any.generateExample = function(grammar, examples, inSyntacticContext, actuals) {
+  return {example: String.fromCharCode(Math.floor(Math.random() * 255))};
+};
+
+function categorizeExamples(examples) {
+  var examplesNeeded = examples.filter(function(example) {
+                                  return example.hasOwnProperty('examplesNeeded');
```

Comment:
Nit: We don't usually indent like this. To be consistent with other code, it should probably be:

```js
var examplesNeeded = examples.filter(function(ex) {
  return ex.hasOwnProperty('examplesNeeded');
}).map(function(ex) {
  return ex.examplesNeeded;
});
```